src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -143,18 +144,18 @@ State Propagator::calculateStateAt(const State& aState, const Instant& anInstant
         throw ostk::core::error::runtime::Undefined("Propagator");
     }
 
-    const State state = {
-        aState.accessInstant(),
-        aState.inFrame(Propagator::IntegrationFrameSPtr).extractCoordinates(this->coordinatesBrokerSPtr_->getSubsets()),
-        Propagator::IntegrationFrameSPtr,
-        this->coordinatesBrokerSPtr_,
-    };
+    const State solverInputState = this->stateBuilder_.reduce(aState.inFrame(Propagator::IntegrationFrameSPtr));
 
-    return numericalSolver_.integrateTime(
-        state,
+    const State solverOutputState = numericalSolver_.integrateTime(
+        solverInputState,
         anInstant,
-        Dynamics::GetSystemOfEquations(this->dynamicsContexts_, state.accessInstant(), Propagator::IntegrationFrameSPtr)
+        Dynamics::GetSystemOfEquations(
+            this->dynamicsContexts_, solverInputState.accessInstant(), Propagator::IntegrationFrameSPtr
+        )
     );
+
+    const StateBuilder outputStateBuilder(aState);
```

Comment:
```suggestion
    const StateBuilder outputStateBuilder = {aState};
```
Nit: Just for consistency